duckduckgo/iOS
# Add GitHub Actions workflow for running unit tests

Task/Issue URL: https://app.asana.com/0/414709148257752/1202223512426983/f

**Description**:
* Added a GitHub Action workflow with two jobs run in parallel: unit tests and swiftlint
* Unit tests job is run on PRs and direct pushes to develop, SwiftLint is run only on PRs
* Unit tests workflow runs on MacOS 12
* SwiftLint is only run on Swift files changed in the PR
* Both jobs would add inline annotations to the PR in places where a test case fails or a linter rule is violated.



**Steps to test this PR**:
1. Verify that actions runs pass (consult [previous runs](https://github.com/duckduckgo/iOS/actions/workflows/pr.yml) if you like)
1. Feel free to commit a broken test or SwiftLint rule breach to this branch and observe the outcome



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
